classes/class-ledyer-confirmation.php

```diff
@@ -33,27 +33,25 @@ public function actions() {
 	 */
 	public function confirm_order() {
 
+		$ledyer_purchase_complete = filter_input( INPUT_GET, 'lco_purchase_complete', FILTER_SANITIZE_STRING );
 		$ledyer_confirm = filter_input( INPUT_GET, 'lco_confirm', FILTER_SANITIZE_STRING );
 		$order_key    = filter_input( INPUT_GET, 'key', FILTER_SANITIZE_STRING );
 		$ledyer_pending    = filter_input( INPUT_GET, 'lco_pending', FILTER_SANITIZE_STRING );
 
-		if ( empty( $ledyer_confirm ) || empty( $order_key ) ) {
+		if ( empty( $ledyer_confirm ) || empty( $order_key ) || empty( $ledyer_purchase_complete) ) {
 			return;
 		}
 		$order_id = wc_get_order_id_by_order_key( $order_key );
 		$order    = wc_get_order( $order_id );
 
 		$order_pending_order = ( 'yes' === $ledyer_pending ) ? true : false;
+		$should_confirm_order = ( 'yes' === $ledyer_purchase_complete ) ? true : false;
 
-		Logger::log( $order_id . ': Confirmation endpoint hit for order.' );
-
-		if ( empty( $order->get_date_paid() ) ) {
-
-			Logger::log( $order_id . ': Confirm the Ledyer order from the confirmation page.' );
+		Logger::log( $order_id . ': Confirm the Ledyer order from the confirmation page.' );
 
+		if ( $should_confirm_order ) {
 			// Confirm the order.
 			wc_ledyer_confirm_ledyer_order( $order_id, $order_pending_order );
-
```

Comment:
This condition prevents confirmation and didn't make sense since we  should only trust the event coming from Ledyer BE in order to continue confirming the order, so changed it a bit